http.go

```diff
@@ -189,53 +189,64 @@ func TracedHTTPHandler(
 	interceptor ...TracedHTTPInterceptor,
 ) http.Handler {
 	mux, muxFound := h.(*http.ServeMux)
-
 	fn := func(w http.ResponseWriter, r *http.Request) {
-		tracer := opentracing.GlobalTracer()
-		parentCtx, _ := tracer.Extract(core.HTTPHeaders, core.HTTPHeadersCarrier(r.Header))
-
-		config := optioniallyInterceptHTTP(interceptor, r)
-
-		var op string
-		if config.OperationName != "" {
-			op = config.OperationName
-		} else {
-			op = httpOperationName(mux, muxFound, r)
-		}
-		opts := []opentracing.StartSpanOption{
-			ChildOf(parentCtx),
-			ext.SpanKindServer(),
-			ext.Component("ctrace.TracedHttpHandler"),
-			ext.HTTPRemoteAddr(r.RemoteAddr),
-			ext.HTTPMethod(r.Method),
-			ext.HTTPUrl(r.URL.String()),
-			ext.HTTPUserAgent(r.UserAgent()),
+		ignored := false
+		for _, v := range interceptor {
+			config := v(r)
+			if config.IgnoredPaths != nil {
+				ignored = config.IgnoredPaths.MatchString(r.URL.String())
+			}
 		}
 
-		if config.OperationName != "" {
-			op = config.OperationName
-		}
-		if len(config.Tags) > 0 {
-			opts = append(opts, config.Tags...)
-		}
-		debug("TracedHttpHandler: StartSpan(%s)", op)
-
-		span := tracer.StartSpan(op, opts...)
-		ctx := span.Context()
-
-		ri := responseInterceptor{
-			tracer: tracer,
-			span:   span,
-			ctx:    ctx.(core.SpanContext),
-			writer: w,
-		}
-
-		if parentCtx != nil {
-			ri.parentCtx = parentCtx.(core.SpanContext)
+		if ignored {
+			h.ServeHTTP(w, r)
```

Comment:
add a return after this and then you can remove the else below and reduce nesting...